chain-api/src/types/dtos.spec.ts

```diff
@@ -187,13 +190,61 @@ describe("ChainCallDTO", () => {
     const dto = new TestDto();
     dto.amounts = [new BigNumber("78.9")];
     dto.signing = SigningScheme.TON;
-    expect(dto.signature).toEqual(undefined);
+    expect(dto.signatures).toEqual(undefined);
 
     // When
     dto.sign(pair.secretKey.toString("base64"));
 
     // Then
-    expect(dto.signature).toEqual(expect.stringMatching(/.{50,}/));
-    expect(dto.isSignatureValid(pair.publicKey.toString("base64"))).toEqual(true);
+    expect(dto.signatures).toHaveLength(1);
+    expect(dto.signatures![0].signature).toEqual(expect.stringMatching(/.{50,}/));
+    expect(dto.isSignatureValid(dto.signatures![0].signature!, pair.publicKey.toString("base64"))).toEqual(
+      true
+    );
+  });
+
+  it("should reject signatures with different schemes", async () => {
+    const { privateKey } = genKeyPair();
+    const dto = new TestDto();
+    dto.amounts = [new BigNumber("1")];
+
+    dto.sign(privateKey);
+
+    const tonPair = await signatures.ton.genKeyPair();
+    dto.signing = SigningScheme.TON;
+
+    expect(() => dto.sign(tonPair.secretKey.toString("base64"))).toThrow(ValidationFailedError);
+  });
+
+  it("should validate signatures using stored signing params", () => {
+    const { privateKey } = genKeyPair();
+    const dto = new TestDto();
+    dto.prefix = "foo";
+    dto.amounts = [new BigNumber("1")];
+
+    dto.sign(privateKey);
+
+    dto.signing = SigningScheme.TON;
+    dto.prefix = "bar";
+
+    expect(dto.isSignatureValid(dto.signatures![0])).toEqual(true);
+  });
+
+  it("should convert legacy single signature", () => {
+    const dto = new TestDto();
+    dto.signature = "legacy";
+    dto.signerPublicKey = "pk";
+
+    convertLegacySignatures(dto);
+
+    expect(dto.signatures).toEqual([
+      {
+        signature: "legacy",
+        signerPublicKey: "pk",
+        signerAddress: undefined,
+        signing: undefined,
+        prefix: undefined
+      }
+    ]);
```

Comment:
we should have multiple tests:
1. covering the signing with multiple keys, to be sure the target object structure is correct and consistent
2. covering the validation whether the object has a single signature or signatures array
3. verifying multi signatures